implementations/rust/ockam/ockam_executor/src/executor.rs

```diff
@@ -177,6 +214,19 @@ impl Task {
     fn allocate(future: impl Future + 'static) -> Box<Task> {
         Box::new(Node {
             id: TaskId::new(),
+            _name: "Task",
+            future: UnsafeCell::new(async {
+                // task terminating
+                future.await;
+            }),
+            // TODO future: Box::pin(future),
```

Comment:
In the mean time, is there a reason to not change the arg to `impl Future + Unpin + 'static`?

Reply:
I did try this now but the `Unpin` trait bound ends up bubbling into `ockam_node` with sundry wreckage ensuing. 😕 

i.e. When we introduce:

```rust
impl Task {
    fn allocate(future: impl Future + Unpin + 'static) -> Box<Task> {]
   ...
```
…we end up with things like:

```rust
   Compiling ockam_node v0.41.1-dev (/Users/antoine/Ockam/ockam.git/implementations/rust/ockam/ockam_node)
error[E0277]: `F` cannot be unpinned
  --> /Users/antoine/Ockam/ockam.git/implementations/rust/ockam/ockam_node/src/executor.rs:78:30
   |
78 |         let _join = rt.spawn(future);
   |                        ----- ^^^^^^ the trait `Unpin` is not implemented for `F`
   |                        |
   |                        required by a bound introduced by this call
   |
   = note: consider using `Box::pin`
note: required by a bound in `Runtime::spawn`
  --> /Users/antoine/Ockam/ockam.git/implementations/rust/ockam/ockam_executor/src/runtime.rs:55:21
   |
55 |         F: Future + Unpin + Send + 'static,
   |                     ^^^^^ required by this bound in `Runtime::spawn`
help: consider further restricting this bound
   |
74 |         F: Future + Send + 'static + core::marker::Unpin,
   |                                    +++++++++++++++++++++
```

Ideally I shouldn't have to rewrite `ockam_node` methods to make this work so I still have some way to go to attain  full grokking of the many nuances here! ☺️

Reply:
Ah, oof. My concern with ignoring it is that it's often a real crash or UB — same as using memory from a functions stack frame after returning from it...

We have some other dodgy unpins here, and this crate is still WIP. Needs to be revisited later, tho.